packages/react-native-renderer/src/legacy-events/SyntheticEvent.js

```diff
@@ -77,6 +77,11 @@ function SyntheticEvent(
   this._dispatchListeners = null;
   this._dispatchInstances = null;
 
+  // React Native Event polyfill - enable Event proxying calls to SyntheticEvent
+  if (nativeEvent.setSyntheticEvent) {
```

Comment:
I'm not sure what this checks for, but I worry about potential deopts here. This is very hot code. If we're checking non-existent property, if I recall correctly it can trigger make it slower. I'm also not sure I understand the purpose of "binding" a native event object back to this one. For example, in DOM we don't need to do that. What's different?

Reply:
We have two APIs we're trying to support now - the props-based one that takes a SyntheticEvent, and the W3C-compliant EventEmitter one that expects an Event/CustomEvent. If a CustomEvent is passed through it gets wrapped in SyntheticEvent by React, but then we actually want calls on one to impact the other, so we need some mechanism to keep them synced. 

My approach is admittedly pretty brute-force. There might be a cleaner way to keep the two in sync.